### ISSUE TITLE
Create validation test for datasets

### ISSUE THREAD

**Issue by seba-1511** [NervanaSystems/neon#57]
Having a way to create dataset splits (eg, validation split from the train set) in the YAML would be useful for reporting validation metrics, and to avoid over-fitting the test set when using spearmint.

**Comment by scttl** [comment]
Hi Séb,

neon already supports the notion of validation dataset partitions, and typically the user would divide their dataset into train/test/validation prior to using neon.

That being said I agree that its occasionally useful to take an existing train/test divided dataset and have the ability to carve some of the training set cases out for validation purposes.  This functionality already exists in neon for datasets that utilize the BatchWriter class like `imageset.py`.  We could do a better job documenting it, but there is a dataset parameter called `validation_pct` that can be passed to this dataset and during the first `load()` call will be used to randomly select a portion of the training cases to use for validation purposes.

**Comment by scttl** [comment]
Closing this as `validation_pct` functionality has been merged in as part of #60
